lib/analist/checker.rb

```diff
@@ -25,9 +25,12 @@ def check_children(node)
     def check_send(node)
       return [Analist::DecorateWarning.new(node)] if node.annotation.hint ==
                                                      Analist::ResolveLookup::Hint::Decorate
-      return [] if node.annotation.return_type[:type] == Analist::AnnotationTypeUnknown
 
       receiver, _method_name, *args = node.children
+      errors = [check(receiver), args.flat_map { |arg| check(arg) }].compact.flatten
```

Comment:
I think the second 'flatten' is not needed, because you are already doing a flat_map on your args array. The array should already be flatten.